### DISCUSSION TITLE
Need help in building tinamcs together with Hugo in GitHub Actions

### DISCUSSION THREAD

**Discussion by leafarafael** [tinacms/tinacms#3586] [Help]
What's the correct way of using tinacms build in Github actions ?
I've tried using run:  tinacms build && hugo --minify but I got an error "tinacms: command not found".

on:
  push:
    branches:
    - main

jobs:
  build_and_deploy:
    name: build and build_and_deploy
    runs-on: ubuntu-latest

    steps:
    - name: checkout
      uses: actions/checkout@v2

    - name: setup hugo
      uses: peaceiris/actions-hugo@v2
      with:
          hugo-version: 'latest'
          extended: true

    - name: build      
      run:  hugo --minify

    - name: setup firebase and deploy
      uses: FirebaseExtended/action-hosting-deploy@v0
      with:
        repoToken: '${{ secrets.GITHUB_TOKEN }}'
        firebaseServiceAccount: '${{ secrets.FIREBASE_SERVICE_ACCOUNT_DARALULOOM_BANIYAS }}'
        channelId: live
        projectId: daraluloom-baniyas
      env:
        FIREBASE_CLI_PREVIEWS: hostingchannels

**Comment by kldavis4** [comment]
Please review this thread in discord: https://discord.com/channels/835168149439643678/835172431836479508/1065857116205748354 - there is an example github actions workflow that similar to what you will need to do for hugo.

**Reply by jamespohalloran** [reply]
I made some updates to this doc today, as there were some issues reported [here](https://github.com/tinacms/tina.io/issues/1487) 

https://tina.io/docs/tina-cloud/connecting-site/
Does that look correct? @kldavis4

**Reply by leafarafael** [reply]
I tried this and I got this error message

Run tinacms build
   tinacms build
  shell: /usr/bin/bash -e ***0***
  env:
    CLIENT_ID: ***
    TINA_TOKEN: ***
/home/runner/work/_temp/bd250661-e308-42b5-8c34-2d8f7117f089.sh: line 1: tinacms: command not found
Error: Process completed with exit code 127.

**Reply by leafarafael** [reply]
@kldavis4 @jamespohalloran 

I found out what seems to work when building tinacms and hugo in Github Actions.

1. I installed react and react-dom in my visual studio project.
2. Setup the Node. js in yml file then install all dependency using npm install. (Tried npm install tinacms but got error)
3. Building site using npm run build (package.json script is "build": "tinacms build && hugo")

`on:
  push:
    branches:
    - main

jobs:
  build_and_deploy:
    name: build and build_and_deploy
    runs-on: ubuntu-latest
    
    steps:
    - name: checkout
      uses: actions/checkout@v2    

    - name: Set up Node.js
      uses: actions/setup-node@v2
      with:
        node-version: '14.x'

    - name: Install dependencies
      run: npm install

    - name: Build site
      run: npm run build

    - name: setup firebase and deploy
      uses: FirebaseExtended/action-hosting-deploy@v0
      with:
        repoToken: '${{ secrets.GITHUB_TOKEN }}'
        firebaseServiceAccount: '${{ secrets.FIREBASE_SERVICE_ACCOUNT_DARALULOOM_BANIYAS }}'
        channelId: live
        projectId: daraluloom-baniyas
      env:
        FIREBASE_CLI_PREVIEWS: hostingchannels
`
